lib/pbench/agent/utils.py

```diff
@@ -390,3 +393,36 @@ def collect_local_info(pbench_bin):
         hostdata[arg] = cp.stdout.strip() if cp.stdout is not None else ""
 
     return (version, seqno, sha1, hostdata)
+
+
+class LocalRemoteHost:
+    """Dynamically build a list of all the local IP addresses so that a simple
+    simple query interface provides the answer to the question, "is the given
+    host name or IP address remote?"
+    """
+
+    def __init__(self):
+        """Build up a list of local IP addresses from all the interfaces.
+        """
+        local_ips_l = []
+        for ifaceName in interfaces():
+            for ifaddr in ifaddresses(ifaceName).setdefault(AF_INET, [{"addr": None}]):
+                ip_addr = ifaddr["addr"]
+                if ip_addr is None:
+                    continue
+                local_ips_l.append(ip_addr)
+        self.local_ips = frozenset(local_ips_l)
+
+    def is_remote(self, host_name):
+        """Returns True if the given host name (or IP address) is remote else
+        returns False.
+        """
```

Comment:
It's a bit pedantic, but I suggest that this function returns `True` if the given host name or IP address is not recognized as referring to the local host.  (We cannot really tell whether it is remote or not without requesting the routing information, and, even then, it might be routed back to the current host...so, all that we can tell is that we don't think it's us.)